### PR TITLE
delete numberOfEggsFrame 

### LINKED ISSUE #13: Remove `numberOfEggsFrame` field from the `Hive` class and use `eggsFrames.size()` instead
Keeping a separate variable to store the `numberOfEggsFrame` requires unnecessary extra work and can easily get out of sync with the real number of `numberOfEggsFrame` and cause bugs, for example in the `Apiary#hibernate` you set `hive.setNumberOfEggsFrame(4);` even though the real number of eggFrames could be different because of previous changes.

### PR DESCRIPTION
**Pull request by MarianBastiurea**
delete numberOfEggsFrame

### CODE REVIEW

File: src/main/java/com/marianbastiurea/lifeofbees/MainLifeOfBees.java [hunk 1]
```diff
@@ -32,18 +32,18 @@ public static void main(String[] args) {
         for (int i = 1; i < numberOfStartingHives + 1; i++) {
             int ageOfQueen = random.nextInt(1, 6);
             int numberOfHoneyFrame = random.nextInt(3, 4);
-            int numberOfEggsFrame = random.nextInt(3, 4);
             int randomNumberOfEggs = random.nextInt(4500, 5000);
             double kgOfHoney = random.nextDouble(2.5, 3);
             List<EggsFrame> eggsFrames = new ArrayList<>();
-            for (int j = 1; j < numberOfEggsFrame; j++) {
+             random.nextInt(3, 4);
```
Comment (andreiursudev) [inline src/main/java/com/marianbastiurea/lifeofbees/MainLifeOfBees.java]: This line is unnecessary, remove it.

**Review by andreiursudev — CHANGES_REQUESTED** [review]
Remove `private int numberOfEggsFrame;` from `Hive` class.

File: src/main/java/com/marianbastiurea/lifeofbees/MainLifeOfBees.java (hunk 1 above)
Comment (MarianBastiurea) [reply src/main/java/com/marianbastiurea/lifeofbees/MainLifeOfBees.java]: was removed

Comment (MarianBastiurea) [reply src/main/java/com/marianbastiurea/lifeofbees/MainLifeOfBees.java]: done

File: src/main/java/com/marianbastiurea/lifeofbees/Apiary.java [hunk 2]
```diff
@@ -63,24 +63,22 @@ public void splitHive(Hive hive) {
 
         List<Hive> newHives = new ArrayList<>();
 
-        if (hive.getNumberOfEggsFrame() == 6 && !hive.isItWasSplit()) {
+        if (hive.getEggsFrames().size() == 6 && !hive.isItWasSplit()) {
             System.out.println("Now old and new frames are full. Hive will be split in two hives.");
             Hive newHive = new Hive( this,this.getNumberOfHives() + 1,
-                    true, true, hive.getNumberOfHoneyFrame(), hive.getNumberOfEggsFrame(),
+                    true, true, hive.getNumberOfHoneyFrame(), hive.getEggsFrames().size(),
```
Comment (andreiursudev) [inline src/main/java/com/marianbastiurea/lifeofbees/Apiary.java]: You don't need to pass `hive.getEggsFrames().size()` anymore. Remove it.

**Review by andreiursudev — CHANGES_REQUESTED** [review]

File: src/main/java/com/marianbastiurea/lifeofbees/Apiary.java (hunk 2 above)
Comment (MarianBastiurea) [reply src/main/java/com/marianbastiurea/lifeofbees/Apiary.java]: done

**Review by andreiursudev — APPROVED** [review]

**Pull request merged**